bindgen/codegen/mod.rs

```diff
@@ -3121,7 +3122,10 @@ impl CodeGenerator for Enum {
             // The custom derives callback may return a list of derive attributes;
             // add them to the end of the list.
             let custom_derives = ctx.options().all_callbacks(|cb| {
-                cb.add_derives(&crate::callbacks::DeriveInfo { name: &name })
+                cb.add_derives(&crate::callbacks::DeriveInfo { 
+                    name: &name,
+                    kind: None,
```

Comment:
Continuation of https://github.com/rust-lang/rust-bindgen/pull/2328/files#r1043753788

And here I'd use `kind: WhateverKind::Enum` as this is the `CodeGenerator` implementation for `Enum`.

In that way you don't need the `Option` wrapper either.